### ISSUE TITLE
Blog plugin should raise warning, not error when related link target does not exist

### ISSUE THREAD

**Issue by alexvoss** [squidfunk/mkdocs-material#6869]
### Context

Building blogs with Insider edition and related links.

### Bug description

When building a blog with related links that do not have a valid target (page or anchor), the blog plugin aborts the build with an error. Instead, it should create a warning so that in non-strict mode the build can finish.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#adding-related-links
- https://www.mkdocs.org/user-guide/configuration/#strict

### Reproduction

[9.5.12+insiders.4.53.0-blog-related-links-should-respect-strictnon-strict.zip](https://github.com/squidfunk/mkdocs-material/files/14482360/9.5.12%2Binsiders.4.53.0-blog-related-links-should-respect-strictnon-strict.zip)

### Steps to reproduce

1. with Insiders, build reproduction project, observe build fail
2. change the link in the single blog post to a non-existing page instead of non-existing anchor, observe same behaviour

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting. We can investigate whether we can change this to a warning, but I'm not sure if this is easily achievable. Please note that this is currently not a priority and we're considering it an edge case, but we can look into it for sure ☺️

**Comment by squidfunk** [comment]
Fixed in [`519170d`](https://github.com/squidfunk/mkdocs-material-insiders/commit/519170d5b90b01597716c683441630c58bf412ff). Unresolvable links will now just be left as-is and a warning is printed:

```
INFO    -  Building documentation...
INFO    -  Cleaning site directory
WARNING -  Error reading metadata of post 'blog/posts/test.md' in 'docs':
           Couldn't find anchor 'does-not-exist' in 'index.md'
WARNING -  Error reading metadata of post 'blog/posts/test.md' in 'docs':
           Couldn't find file for link 'foo.md'
INFO    -  Documentation built in 0.15 seconds
```

**Comment by squidfunk** [comment]
Released as part of [9.5.13+insiders-4.53.1](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.53.1).
